app/src/main/java/org/openobservatory/ooniprobe/activity/MeasurementDetailActivity.java

```diff
@@ -203,19 +205,36 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             }
         }
         assert detail != null && head != null;
-        getSupportFragmentManager().beginTransaction()
-                .replace(R.id.footer, ResultHeaderDetailFragment.newInstance(
-                        true,
-                        null,
-                        null,
-                        measurement.start_time,
-                        measurement.runtime,
-                        false,
-                        measurement.result.network.country_code,
-                        measurement.result.network))
-                .replace(R.id.body, detail)
-                .replace(R.id.head, head)
-                .commit();
+        if (measurement.rerun_network!=null && !measurement.rerun_network.isEmpty()){
+            Network network = new Gson().fromJson(measurement.rerun_network,Network.class);
+            getSupportFragmentManager().beginTransaction()
+                    .replace(R.id.footer, ResultHeaderDetailFragment.newInstance(
+                            true,
+                            null,
+                            null,
+                            measurement.start_time,
+                            measurement.runtime,
+                            false,
+                            network.country_code,
+                            network))
+                    .replace(R.id.body, detail)
+                    .replace(R.id.head, head)
+                    .commit();
+        }else {
+            getSupportFragmentManager().beginTransaction()
+                    .replace(R.id.footer, ResultHeaderDetailFragment.newInstance(
+                            true,
+                            null,
+                            null,
+                            measurement.start_time,
+                            measurement.runtime,
+                            false,
+                            measurement.result.network.country_code,
+                            measurement.result.network))
+                    .replace(R.id.body, detail)
+                    .replace(R.id.head, head)
+                    .commit();
+        }
```

Comment:
```suggestion
        var net = measurement.result.network;
        var cc = measurement.result.network.country_code;
        if (measurement.rerun_network != null && !measurement.rerun_network.isEmpty()) {
            Network network = new Gson().fromJson(measurement.rerun_network,Network.class);
            net = network;
            cc = network.country_code;
        }
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.footer, ResultHeaderDetailFragment.newInstance(
                       true,
                       null,
                       null,
                       measurement.start_time,
                       measurement.runtime,
                       false,
                       cc,
                       net))
                .replace(R.id.body, detail)
                .replace(R.id.head, head)
                .commit();
```

I think this is more compact and less painful in case of future refactorings